src/patchelf.cc

```diff
@@ -2291,6 +2542,25 @@ static int mainWrapped(int argc, char * * argv)
         else if (arg == "--add-debug-tag") {
             addDebugTag = true;
         }
+        else if (arg == "--rename-dynamic-symbols") {
+            renameDynamicSymbols = true;
+            if (++i == argc) error("missing argument");
+
+            std::ifstream infile(argv[i]);
+            if (!infile) error(fmt("Cannot open map file ", argv[i]));
+
+            std::string from, to;
+            while (true)
+            {
+                if (!(infile >> from))
+                    break;
+                if (!(infile >> to))
+                    error("Odd number of symbols in map file");
```

Comment:
I wonder if this should be a bit more strict. Currently both formats are accepted:

```
symbol
symbol_2
```

and

```
symbol symbol_2
```

Reply:
Yes, perhaps it was just lazyness on my part.
Requiring symbol pairs to be in one line is better because we can error on a bad map file and say the line that is wrong.

Would you agree that we should also error if an @ is found in the symbol name and mention that this does not work with symbol versions?
I saw you added it to the help, and during development I made this mistake myself (just taking the output of NM) so it would be an useful check.

I'll get to it

Reply:
Changed it such that we read two names per line.
If the line is missing the second element, we get:
```
patchelf: <filename>:<line>: Map file line is missing the second element
```
If we have two symbols, we error out when we find the second:
```
patchelf: <filename>:<line>: Name '<name>' appears twice in the map file
```
If either of the names contain "@", we have:
```
patchelf: <filename>:<line>: Name pair contains version tag: <first name> <second name>
```